client/src/components/TeacherRegistration.jsx

```diff
@@ -0,0 +1,63 @@
+// src/components/TeacherRegistration.js
+import React from "react";
+
+const handleSubmit = async (e) => {
+  e.preventDefault();
+
+  const data = {
+    username: document.getElementById("username").value,
+    password: document.getElementById("password").value,
+  };
```

Comment:
You shouldn't be accessing the DOM directly, the information from the submission of the form should be accessible from the `e` object under `e.target`. Just make sure that you do a null checks beforehand to make sure that both target and the attributes that you are trying to access are actually there.